src/app/core/components/add-marios-form/add-marios-form.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, Input } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MariosService } from '../../services/marios.service';
+import { Router } from '@angular/router';
+
+import { Employee } from '../../models/employee.model';
+
+@Component({
+  selector: 'app-add-marios-form',
+  templateUrl: './add-marios-form.component.html',
+  styleUrls: ['./add-marios-form.component.scss'],
+})
+export class AddMariosFormComponent {
+  @Input() public maxTitleLength = 128;
+  @Input() public maxMessageLength = 255;
+
+  form: FormGroup = new FormGroup({
+    receiversId: this.createFormControlWithValidation([], []),
+    reaction: this.createFormControlWithValidation('', []),
+    title: this.createFormControlWithValidation('', [
+      Validators.required,
+      Validators.minLength(1),
+      Validators.maxLength(this.maxTitleLength),
+    ]),
+    message: this.createFormControlWithValidation('', [
+      Validators.required,
+      Validators.minLength(1),
+      Validators.maxLength(this.maxMessageLength),
+    ]),
+  });
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private mariosService: MariosService,
+    private router: Router
+  ) {}
+
+  private createFormControlWithValidation(initialValue: any, validators: any) {
+    return this.formBuilder.control(initialValue, validators);
+  }
+
+  onSubmit() {
+    if (this.form.valid) {
+      this.addMarios();
+    } else {
+      this.markFormControlsAsTouched();
+    }
+  }
+
+  private extractEmployeeIds() {
+    this.form.value.receiversId = this.form.value.receiversId.map(
+      (employee: Employee) => employee.id
+    );
+  }
+
+  private addMarios() {
+    this.extractEmployeeIds();
```

Comment:
FYI: It could be also 

```js
const payload = {...this.form.value, receiversId:  this.form.value.receiversId.map(
      (employee: Employee) => employee.id
    );}
 //or
const payload = {...this.form.value, receiversId: extractEmployeeIds()}

  this.mariosService.addMarios(payload).subscribe(
```